grpcreflect.go

```diff
@@ -314,31 +314,39 @@ type ExtensionResolver interface {
 }
 
 type fileDescriptorNameSet struct {
-	names map[protoreflect.FullName]struct{}
+	names map[string]struct{}
 }
 
-func (s *fileDescriptorNameSet) Insert(fd protoreflect.FileDescriptor) {
+func (s *fileDescriptorNameSet) Insert(path string) {
```

Comment:
nit: `fileDescriptorNameSet` use to take an fd. It looks like it changed to a path for testing. Maybe could keep the fd and the tests reference the fd too to make it clear that FullName wasn't unique. Otherwise this is a generic string set type.

Reply:
Perhaps this could be a set of `type pather interface { Path() string }`? I do like the fact that the existing code keeps the set type a little higher-level - it'd be nice to preserve that. In tests, we could implement the `pather` interface with `type path string`. That also leaves a clear place to document that `FullName` isn't unique.

Reply:
Alright. I just changed the signatures back to `protoreflect.FileDescriptor` and then added a `dummyFile` implementation that the test can use for this.